aptos-move/move-examples/data_structures/sources/smart_tree.move

```diff
@@ -0,0 +1,971 @@
+/// Type of large-scale search trees.
+///
+/// It internally uses BTree to organize the search tree data structure for keys. Comparing with
```

Comment:
Just to understand the context. The whole point of using a tree data structure here is to make it iterable, right?

Reply:
Make it iterable and searchable (in logN time)